foundation/foundation-configuration-store/src/test/java/com/boozallen/aissemble/configuration/mutating/webhook/MutatingWebhookSteps.java

```diff
@@ -12,30 +12,59 @@
 
 import static io.restassured.RestAssured.given;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
 import io.fabric8.kubernetes.api.model.GroupVersionKind;
 import io.fabric8.kubernetes.api.model.GroupVersionResource;
-import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
-import io.fabric8.kubernetes.api.model.Pod;
-import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
 import io.fabric8.kubernetes.api.model.admission.v1.AdmissionRequest;
 import io.fabric8.kubernetes.api.model.admission.v1.AdmissionReview;
 import io.fabric8.kubernetes.api.model.admission.v1.AdmissionReviewBuilder;
 import io.restassured.response.ValidatableResponse;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
 public class MutatingWebhookSteps {
+    private static final String expectedInjectedValue = "env-access-key-id";
+    private final ObjectMapper objectMapper = new ObjectMapper();
     private ValidatableResponse response;
     private AdmissionReview admissionReviewRequest;
-    private ObjectMapper objectMapper = new ObjectMapper();
+    private ConfigMap configMap;
+    private Map<String, String > configMapData;
+    private String responsePatch;
+
+    @Given("a ConfigMap definition that contains the substitution key exists")
+    public void aConfigMapDefinitionThatContainsTheSubstituionKeyExists() {
+        final String propertyKey = "groupName=aws-credentials;propertyName=AWS_ACCESS_KEY_ID";
+        configMapData = new HashMap<>();
+        configMapData.put("AWS_ACCESS_KEY_ID", ConfigMutatingWebhook.CONFIG_STORE_INJECT_START.replaceAll("\\\\", "") + propertyKey + ConfigMutatingWebhook.CONFIG_STORE_INJECT_END);
```

Comment:
A: Instead of writing the string literal for INJECT_START as a regex and then trying to unescape it here, you could store the exact string in the literal and then use `Pattern.quote` anywhere it needs to be passed in as a regex.